include/ideep/operators/conv.hpp

```diff
@@ -1416,28 +1470,37 @@ struct convolution_backward_data : public dnnl::convolution_backward_data {
     auto diff_src_desc = 
         tensor::desc(diff_src_dims, diff_dst_desc.get_data_type(), format_tag);
 
+    auto op_attr = attr;
+    op_attr.set_scratchpad_mode(dnnl::scratchpad_mode::user);
+
     auto forward_hints =
         convolution_forward::get_primitive_desc</*with_bias=*/false>(
             diff_src_desc, weights_desc, tensor::desc(), diff_dst_desc, strides,
-            dilates_, padding_l, padding_r);
-
-    auto op_attr = dnnl::primitive_attr();
-    op_attr.set_scratchpad_mode(dnnl::scratchpad_mode::user);
+            dilates_, padding_l, padding_r, op_attr);
 
     auto pd = primitive_desc(
         {aalgorithm, diff_src_desc, weights_desc, diff_dst_desc, strides,
          dilates_, padding_l, padding_r}, op_attr, aengine, forward_hints);
 
     auto expected_diff_dst = diff_dst.reorder_if_differ_in(pd.diff_dst_desc());
     auto expected_weights = weights_.reorder_if_differ_in(pd.weights_desc());
-    diff_src.reinit_if_possible(pd.diff_src_desc());
+
+    auto expected_diff_src_desc = pd.diff_src_desc();
+    tensor expected_diff_src;
+    // diff_src not init in FW or has same desc with expected desc.
+    if (diff_src.is_empty() || diff_src.get_desc() == expected_diff_src_desc) {
+      diff_src.reinit_if_possible(expected_diff_src_desc);
+      expected_diff_src = diff_src;
+    } else {
+      expected_diff_src.init(expected_diff_src_desc);
```

Comment:
nit: need to sync result back to diff_src in this case.

Reply:
OK It's fixed